src/components/radio-button-group/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as RadioGroup from '@radix-ui/react-radio-group';
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+
+interface Props {
+  onChange?: (value: string) => void;
+  options: RadioButtonProps[];
+
+  // Group Props
+  defaultValue?: string;
+  disabled?: boolean;
+  id?: string;
+  required?: boolean;
+  value?: string;
+}
+
+export function RadioButtonGroup(props: Readonly<Props>) {
+  const { onChange, options, ...groupProps } = props;
+
+  return (
+    <RadioGroupWrapper onValueChange={onChange} {...groupProps}>
+      {options.map((o) => (
+        <RadioButton groupId={groupProps.id} {...o} />
+      ))}
+    </RadioGroupWrapper>
+  );
+}
+
+/*
+ * Related to the explanation below, we require an aria-label if the label is a ReactNode.
+ * If the label is a string, we use it directly as a fallback if no aria-label is provided
+ */
+type RadioButtonProps = {
+  disabled?: boolean;
+  groupId?: string;
+  value: string;
+} & ({ ariaLabel: string; label: ReactNode } | { ariaLabel?: string; label: string });
+
+function RadioButton(props: Readonly<RadioButtonProps>) {
+  const { ariaLabel, disabled, groupId = 'radio', label, value } = props;
```

Comment:
Should we always have a default groupId?  or just make the group id non optional?

Reply:
Good point, let's make it mandatory